packages/tasit-action/src/TasitAction.js

```diff
@@ -19,117 +22,370 @@ class Utils {
 }
 
 class Subscription {
+  #emitter;
+  #events = [];
+
+  constructor(eventEmitter) {
+    this.#emitter = eventEmitter;
+  }
+
+  removeListener = (eventName, listener) => {
+    this.#events = this.#events.filter(event => {
+      if (event.eventName === eventName && event.listener === listener) {
+        this.#emitter.removeListener(
+          event.wrappedEventName,
+          event.wrappedListener
+        );
+        return false;
+      }
+      return true;
+    });
+  };
+
+  off = (eventName, listener) => {
+    this.removeListener(eventName, listener);
+  };
+
+  listenerCount = eventName => {
+    return this.#events.filter(event => {
+      return event.eventName === eventName;
+    }).length;
+  };
+
+  removeAllListeners = () => {
+    this.#events.forEach(event => {
+      this.removeListener(event.eventName, event.listener);
+    });
+  };
+
+  unsubscribe = () => {
+    this.removeAllListeners();
+  };
+
+  // TODO: Make protected
+  emitErrorEvent = error => {
+    this.#events.forEach(event => {
+      if (event.eventName === "error") {
+        event.listener(error);
+      }
+    });
+  };
+
+  // TODO: Make protected
+  addErrorListener = listener => {
+    const eventName = "error";
+    this.#events.push({
```

Comment:
Maybe the correct name for this array is: `eventListeners` is that what this array is storing.

Reply:
Hmm yes. Let's hold off of any changes based on this thread until we decide on the one listener per event / many listener per event discussion here https://github.com/tasitlabs/TasitSDK/pull/87#discussion_r246035230

Reply:
We've decided on that topic now, so we can make a decision about this thread.

Reply:
I'm okay with your suggestion for the name!

Reply:
Added to new issue https://github.com/tasitlabs/TasitSDK/issues/112

Reply:
Thanks. Should we resolve this or keep it open?

Reply:
I think that TODO here is to rename Map to `eventListeners` if that's correct we should resolve this because #112 is tracking a thread with the same request.